src/mongo/db/pipeline/accumulator_percentile.cpp

```diff
@@ -0,0 +1,190 @@
+            this->digest_size = 1000;
+            }
+        else
+            {
+            this->digest_size = input.getDocument()["digest_size"].getDouble();
+            }
+    }
+
+    // ToDo: Error codes are not accurate. Set better numbers later
+    // ToDo: It might be better evaluations for this part.
+    uassert(6677, "The 'percentile' should be present in the input document.",
+    !input.getDocument()["percentile"].missing());
+
+    uassert(6678, "The 'value' should be present in the input document.",
+    !input.getDocument()["value"].missing());
+
+    this->percentile = input.getDocument()["percentile"].getDouble();
+
+    // ToDo: Choose a better name for input_value and refactor later
+    Value input_value = input.getDocument()["value"];
+
+    switch (input_value.getType()) {
+        case NumberDecimal:
+        case NumberLong:
+        case NumberInt:
+        case NumberDouble:
+            values.push_back(input_value.getDouble());
+            break;
+        default:
+            dassert(!input_value.numeric());
+            return;
+    }
+    
+    if (any_input == false)
+    {
+        digest = mongo::TDigest(this->digest_size);
+        any_input = true;
+    }
+
+    if (values.size() == this->chunk_size){
+        _add_to_tdigest(values);
+        }
+}
+
+intrusive_ptr<Accumulator> AccumulatorPercentile::create(
+    const boost::intrusive_ptr<ExpressionContext>& expCtx) {
+    return new AccumulatorPercentile(expCtx);
+}
+
+Value AccumulatorPercentile::getValue(bool toBeMerged) {
+
+    // To add the remainders
+    if (not values.empty()){
+        _add_to_tdigest(values);
+        }
+
+    if (toBeMerged) {
+        std::vector<Document> centroids;
+
+        for (const auto& centroid: this->digest.getCentroids()) {
+            centroids.push_back(Document{
+                    {"mean", centroid.mean()},
+                    {"weight", centroid.weight()}
+                });
+        };
+
+        return Value(
+            Document{
+                {"centroids", Value(centroids)},
+                {"sum", digest.sum()},
+                {"count", digest.count()},
+                {"max", digest.max()},
+                {"min", digest.min()},
+                {"percentile", this->percentile},
+                {"digest_size", this->digest_size}               
+            }
+        );
+    }
+    return Value(digest.estimateQuantile(this->percentile));
+}
+
+AccumulatorPercentile::AccumulatorPercentile(const boost::intrusive_ptr<ExpressionContext>& expCtx)
+    : Accumulator(expCtx) {
+    // This is a fixed size Accumulator so we never need to update this
+    _memUsageBytes = sizeof(*this);
+}
+
+void AccumulatorPercentile::_add_to_tdigest(std::vector<double> & values){
+    // Sort, Push and Clear the "values" vector in each chunk
+    std::sort(values.begin(), values.end());
+    digest = digest.merge(values);
+    values.clear();
+}
+
+void AccumulatorPercentile::reset() {
+    return;
```

Comment:
Why is the 'reset' not doing anything ?
Since it is there we should implement this method properly